static/flow/views/pi-selector-panel.js

```diff
@@ -380,9 +450,7 @@ var PiSelectorPanel = function(options) {
                 console.log("[DEBUG] Requesting sensor data from ", controller.path);
                 
                 if(!_this.currentlyRecording){
-                   runProgramButton.html('<span class="glyphicon glyphicon-play"></span><span class="deviceRunButtonText">run program on ' + controller.name + '</span>');
-                   runProgramButton.addClass("rungreen");
-                   runProgramButton.removeClass("stopred");
+                   updateProgramButtons(false, false, false);
```

Comment:
Looks like this call is another good refactor :)

Reply:
Agreed.